src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeRecordsAndCompactCtors.java

```diff
@@ -36,7 +36,7 @@ record MyTestRecord2(String string) implements Cloneable { // violation
         }
     }
 
-    record MyTestRecord3(String str, TreeSet treeSet) { // violation
+    record MyTestRecord3(String str, Set treeSet) { // OK
         void foo(HashMap<Integer, Integer> hashMap) { // violation
```

Comment:
Restore this line, it should not be charged

Reply:
Done!